kong/utils_test.go

```diff
@@ -2822,3 +3142,710 @@ func Test_FillPluginsDefaults_NonEmptyDefaultArrayField(t *testing.T) {
 		})
 	}
 }
+
+func Test_ClearUnmatchingDeprecationsSimple(t *testing.T) {
+	tests := []struct {
+		name                     string
+		newPlugin                *Plugin
+		oldPlugin                *Plugin
+		expectedOldPluginCleared Configuration
```

Comment:
Nit: I'd rename this to 

```suggestion
		expectedOldPlugin Configuration
```

the `cleared` suffix implies that this is something that was cleared (removed) from `oldPlugin` rather then being the thing that we expect as is.